app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,10 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = User.find_by(id: current_user)
+    @posts = Post.where(author_id: current_user)
+  end
 
```

Comment:
## [OPTIONAL]

![image](https://user-images.githubusercontent.com/65068771/169437141-1915585c-4678-443e-9be6-1e9321626abc.png)

## 1. 
int your code mode `models/post.rb:line 8...`


```rb
  def update_post_counter
    count = Post.where(author_id: id).count
    user = User.find_by_id(id: author_id)
    user.update(posts_counter: count)
  end
```

- [ ] I would really suggest you solve this error when the post is created. you can solve it by improving the method `update_post_counter` with

```rb
  def update_post_counter
    author.increment!(:posts_counter)
  end
```



## 2
int your code mode `models/comment.rb:line 7...`

```rb
  def update_comment_counter
    count = Comment.where(post_id: id).count
    post = Post.find_by_id(id: post_id)
    post.update(comments_counter: count)
  end
```

- [ ] The same error is in the comment modal also, kindly solve it.  you can solve it with
```rb 
  def update_comment_counter
    post.increment!(:comments_counter)
  end
```
![image](https://user-images.githubusercontent.com/65068771/169437936-b1f604b7-5e90-4ca0-a2ec-b27bd76bc5a1.png)

Reply:
Hi @andersonlebon,

Thank you for your suggestions. I really appreciate them.